islandora_checksum_checker.module

```diff
@@ -213,10 +214,22 @@ function islandora_checksum_checker_validate_checksum($pid) {
 
   // Force the cron to run as user 1 so we don't get locked out of objects.
   $user = user_load(1);
+
   drupal_static_reset('islandora_get_tuque_connection');
-  $tuque = islandora_get_tuque_connection($user);
-  $islandora_object = islandora_object_load($pid);
+  try {
+    if (!$tuque = islandora_get_tuque_connection($user)) {
+      $error = 'Tuque connection failed.';
+      throw new Exception($error);
+    }
+  }
+  catch (Exception $e) {
+    $exceptions[] = $e;
+    // Only invoke the hook on failure.
+    module_invoke_all('islandora_checksum_checker_outcome', $pid, '', array(), '', $exceptions);
+    return FALSE;
+  }
 
+  $islandora_object = islandora_object_load($pid);
```

Comment:
Assuming my first concern gets answered or is unjustified. islandora_object_load($pid); itself should be enough somehow..i mean if the actual user that invokes  ```islandora_checksum_checker_validate_checksum()``` function is user 1. 
Ok. back to the logic: So islandora_object_load() at line 232 has 3 types of responses: the actual fedora object which contains a tuque object , FALSE if the load failed because it was not found and NULL if not accessible because of permissions. So happens that a test on:

var_dump(count(NULL)) leads to int(0)
and
var_dump(count(FALSE)) leads to int(1) and that brings us into the loop even if loading was not possible.

not sure why ```count``` in like 233 instead of an isset().